### PR TITLE
Improve pagination component

### PR DESCRIPTION
**Pull request by gcusnieux**
Related to gravitee-io/issues#6256

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://5ffff84833d7150021078521-wksyhclsgx.chromatic.com)

### CODE REVIEW

**Comment by graviteeio** [comment]
Hey dear reviewer, I'm the Gravitee.io bot :robot:

I just want to tell you that `package.json` has been updated in this PR. 

Some components are based on third party libraries, for integration into a web component, we extract the css from its libraries as assets.

WARNING: When `highlight.js` or `github-markdown-css` are updated, we must run `yarn build` task to update the css.

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 556aefa38db3242a15132a3ff36e828c2bba0338
</p>

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-556aefa
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-556aefa
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-1027d24
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-1027d24
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-c1f034d
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-c1f034d
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-6ef803a
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-6ef803a
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-ed57edd
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-ed57edd
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-c7ab8bc
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-c7ab8bc
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.0-pagination-c27e767
```
or
```shell
yarn add @gravitee/ui-components@3.27.0-pagination-c27e767
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.1-pagination-34a0cc9
```
or
```shell
yarn add @gravitee/ui-components@3.27.1-pagination-34a0cc9
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.1-pagination-c25c075
```
or
```shell
yarn add @gravitee/ui-components@3.27.1-pagination-c25c075
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.1-pagination-a71d7a1
```
or
```shell
yarn add @gravitee/ui-components@3.27.1-pagination-a71d7a1
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.2-pagination-d3ff300
```
or
```shell
yarn add @gravitee/ui-components@3.27.2-pagination-d3ff300
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.2-pagination-e8461d3
```
or
```shell
yarn add @gravitee/ui-components@3.27.2-pagination-e8461d3
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.2-pagination-7a03dcf
```
or
```shell
yarn add @gravitee/ui-components@3.27.2-pagination-7a03dcf
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.2-pagination-5347b00
```
or
```shell
yarn add @gravitee/ui-components@3.27.2-pagination-5347b00
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.2-pagination-e5ed409
```
or
```shell
yarn add @gravitee/ui-components@3.27.2-pagination-e5ed409
```

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.3-pagination-4da8f9f
```
or
```shell
yarn add @gravitee/ui-components@3.27.3-pagination-4da8f9f
```

File: src/organisms/gv-pagination/gv-pagination.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  }
+
+  private get pageSize() {
+    if (this.hasSelect) {
+      return this.data?.size || this.data?.sizes[0];
+    }
+    return null;
+  }
+
+  // @ts-ignore
```
Comment (ThibaudAV) [inline src/organisms/gv-pagination/gv-pagination.ts:267]: still necessary ? one `{ detail }: any` or better type would not be enough ?

File: src/organisms/gv-pagination/gv-pagination.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  private onSubmit(e: CustomEvent<number>) {
+    const page = e.detail;
+    if (page != null && page <= this.data.total_pages && page > 0) {
+      this.goToPage(page);
+    }
+  }
+
+  private onClickToSearch() {
+    if (this.input) {
+      // @ts-ignore
```
Comment (ThibaudAV) [inline src/organisms/gv-pagination/gv-pagination.ts:160]: still necessary ? input is any so it should be good?

**Review by ThibaudAV — APPROVED** [review]
LGTM 

Just a personal impression about `renderPagination`
I find this function not easy to understand quickly. Maybe it should be cut out or add one or two comments. 
but not the subject here

**Review by marcambier — APPROVED** [review]

File: src/organisms/gv-pagination/gv-pagination.ts (hunk 2 above)
Comment (gcusnieux) [reply src/organisms/gv-pagination/gv-pagination.ts]: ✅

File: src/organisms/gv-pagination/gv-pagination.ts (hunk 1 above)
Comment (gcusnieux) [reply src/organisms/gv-pagination/gv-pagination.ts]: ✅

**Comment by gcusnieux** [comment]
> LGTM
> 
> Just a personal impression about `renderPagination` I find this function not easy to understand quickly. Maybe it should be cut out or add one or two comments. but not the subject here

Yes, I agree.. We can create

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

A beta version of this package has been published! 🚀

To use it in your project run the following lines:
```shell
npm install @gravitee/ui-components@3.27.3-pagination-289924a
```
or
```shell
yarn add @gravitee/ui-components@3.27.3-pagination-289924a
```

**Pull request merged**